### PR TITLE
Add checks to ensure distance is strictly less than half of kernel width

### PR DESCRIPTION
**Pull request by MelodyShih**
This is related to #112 .

The error happens when nonuniform points lie on the fine grid. 
The distance between the point and the grid point are exactly `kernelwidth/2` and cause 
`exp(es_beta * (sqrt(1.0 - es_c*x*x)))` evaluates to `nan`. 

I added checks to enforce this does not happen.

### CODE REVIEW

**Comment by ahbarnett** [comment]
Yes, this was needed in FINUFFT too, when the plain exp(sqrt(..)) kernel used:
https://github.com/flatironinstitute/finufft/blob/adcb1d25aa8f3b26aed1956a9dccc12bd8a78eea/src/spreadinterp.cpp#L639
With the vectorized Horner eval, there are no NaNs, so no such check was needed.

Note that your PR has an awful lot of repetition of the check. Could it be rationalized so that the kernel eval and associated check is done in one place only?

You might want to check that tweaks to rounding and wrapping in FINUFFT over the last year have been brought into cuFINUFFT. I can list commits if you need, but have a look in spreadinterp.cpp, basically.
Best, Alex

File: src/2d/spreadinterp2d.cu [hunk 1]
```diff
@@ -565,13 +570,13 @@ void Interp_2d_NUptsdriven(FLT *x, FLT *y, CUCPX *c, CUCPX *fw, int M, const int
 		cnow.y = 0.0;
 		for(int yy=ystart; yy<=yend; yy++){
 			FLT disy=abs(y_rescaled-yy);
-			FLT kervalue2 = evaluate_kernel(disy, es_c, es_beta);
+			FLT kervalue2 = disy<ns/2.0 ? evaluate_kernel(disy, es_c, es_beta):0;
```
Comment (ahbarnett) [inline src/2d/spreadinterp2d.cu:573]: Hi, while this works, I suggest you incorporate this check into evaluate_kernel. This requires adding an extra arg, ns. But then you won't have the same check repeated 10 times as below, which is much harder to maintain...

**Review by ahbarnett — CHANGES_REQUESTED** [review]
Hi Melody - sorry, if you have time, could you tweak this as in my code comment, for cleaner non-repetitive code?  Thanks, Alex

**Comment by MelodyShih** [comment]
Hi Alex, sure, sounds better to do what you suggest. I will update with a new commit. Best regards, Melody

**Review by ahbarnett — APPROVED** [review]

**Pull request merged**